File: kotlin/step/math/main_1002.kt
import kotlin.math.absoluteValue
import kotlin.math.pow
import kotlin.math.sqrt

fun main() {
    val count = readLine()!!.toInt()
    for (i in 1..count){
        val list = readLine()!!.split(' ').map { it.toDouble() }
        val abLen = sqrt((list[3] - list[0]).pow(2) + (list[4] - list[1]).pow(2))
        println(when {
            abLen == 0.0 -> {
                if (list[2] == list[5])
                    -1
                else
                    0
            }
            abLen > list[2] + list[5] || abLen < (list[2] - list[5]).absoluteValue -> 0
            abLen == list[2] + list[5] || abLen == (list[2] - list[5]).absoluteValue -> 1
            else -> 2
        })
    }
}

File: cpp/step/math/main_1002.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
	int count;
	cin >> count;
	for (int i = 0; i < count; ++i) {
		int x1, y1, r1, x2, y2, r2;
		cin >> x1 >> y1 >> r1 >> x2 >> y2 >> r2;

		const double abLen = sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2));
		if (abLen == 0) {
			if (r1 == r2) {
				cout << -1 << endl;
			} else {
				cout << 0 << endl;
			}
		} else if (abLen > r1 + r2 || abLen < abs(r1 - r2)) {
			cout << 0 << endl;
		} else if (abLen == r1 + r2 || abLen == abs(r1 - r2)) {
			cout << 1 << endl;
		} else {
			cout << 2 << endl;
		}
	}
}